### ISSUE TITLE
Incorrect E3031 error reported for SES configuration set

### ISSUE THREAD

**Issue by philipstarkey** [aws-cloudformation/cfn-lint#4484]
### CloudFormation Lint Version

1.50.0

### What operating system are you using?

Mac

### Describe the bug

I'm running `cfn-lint` as a pre-commit hook using `prek`. As of 1.49.0, the following error is now being reported:
```
  E3031 'True' does not match 'true|false'
  mail.yaml:7:9

  E3031 'True' does not match 'true|false'
  email.yaml:9:9
```

The same issue appears if you use `false` in the template as well.

As you can see from the minimal example template below, the values are not `True` (they are `true`). This is not an issue on v1.48.1 and I believe is a regression introduced in v1.49.0 (I've also tested up to v1.50.0 and it is not fixed in those either).

### Expected behavior

When using `true` as a value, it shouldn't tell you that you have "used `True` which doesn't match `true`" given you have actually used what it is telling you to change it to.

### Reproduction template

```yaml
Resources:
  sesconfigurationset:
    Type: AWS::SES::ConfigurationSet
    Properties:
      Name: !Sub "${AWS::StackName}-configuration-set"
      ReputationOptions:
        ReputationMetricsEnabled: true
      SendingOptions:
        SendingEnabled: true
```
